src/arrow_array_schema_proxy.cpp

```diff
@@ -301,24 +301,13 @@ namespace sparrow
         schema().name = private_data->name_ptr();
     }
 
-    [[nodiscard]] std::optional<std::string_view> arrow_proxy::metadata() const
+    [[nodiscard]] std::optional<key_value_view> arrow_proxy::metadata() const
     {
         if (schema().metadata == nullptr)
         {
             return std::nullopt;
         }
-        return std::string_view(schema().metadata);
-    }
-
-    void arrow_proxy::set_metadata(std::optional<std::string_view> metadata)
```

Comment:
Was this method indirectly exposed ot the end user? If so, how can a user set the metadata now?

Reply:
This function is now templated and so moved to the header